### DISCUSSION TITLE
Welcome to TimePicker.PCF Discussions!

### DISCUSSION THREAD

**Discussion by drivardxrm** [drivardxrm/TimePicker.PCF#14] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bakererica** [comment]
I am new to controls and I'm having a hard time getting this one to work.  What kind of column do I need to create on the dataverse table for this to work?  I used whole number because it is the only one that will allow me to select the time picker control but then it won't take the time, it just takes the minutes, no special characters or the am pm.  Thank you for your time.

**Comment by drivardxrm** [comment]
Hi @bakererica,

Thanks for trying out the control, I hope you will like it.

To configure the PCF correctly, you need to supply 2 backing fields (Whole.Number), 1 for the hours (Main field connected to the control) and 1 for the minutes.

![image](https://github.com/drivardxrm/TimePicker.PCF/assets/38399134/2a350e75-b288-4b03-9484-9687fa54a5b7)

Hope this helps

**Comment by bakererica** [comment]
Thank you for responding. I tried that and it will only fill out the
hours.  I'm missing a step somewhere I guess.

**Reply by drivardxrm** [reply]
Are you sure that you output to 2 different fields ?

![image](https://github.com/drivardxrm/TimePicker.PCF/assets/38399134/dc200874-645c-42f9-b86a-d57443815ce7)

**Comment by bakererica** [comment]
Yep, I tried 2 different and then tried to make it the same.  I'll start
fresh again tomorrow and see what I get.   I appreciate you responding!

**Comment by bakererica** [comment]
[image: image.png]

This is what keeps happening.  I hate to keep asking you questions, it's
just that this one component will make everything better!   I have tried
directly binding to the table like shown and without those fields filled in
as well.

[image: image.png]

Thank you for any suggestions!

**Comment by bakererica** [comment]
Doesn't look like the attachments worked.  Here they are. 
![image (1)](https://github.com/drivardxrm/TimePicker.PCF/assets/150947961/14d758f6-c0b0-4711-94df-8d605840aacb)
![image](https://github.com/drivardxrm/TimePicker.PCF/assets/150947961/b9212bbf-c756-4a3d-8858-038928644dbc)

**Reply by drivardxrm** [reply]
Hi @bakererica,

Looking at your configuration 
1- The field called TimePicker would be the one where your hours are saved, If you want to have the hours on another field ex. Hours, just bind the control to this field.

2- You should not use the same field for HourStep and MinuteStep, 
these are optional field and serve for example if you want to have a 15 min step (0, 15, 30, 45)
Actually, you should not put a field here but a manual value if you want to use it. 
If you leave these blank, the step will be 1 by default.

![image](https://github.com/drivardxrm/TimePicker.PCF/assets/38399134/6d739ac1-b7f5-4410-aafa-c0bb406bd736)

Hope this helps,

**Reply by bakererica** [reply]
I have tried that but when I add the component to the TimePicker field (in this case, I created a new test field and removed the component from the other field) it automatically sets the Hour field to the field I'm on.

**Reply by bakererica** [reply]
I got it to work finally! Thank you!!!!!

**Reply by drivardxrm** [reply]
That is how it works.
The field where you bind the control will hold the 'hour' value and the field bound on the Minute value field will hold the minute.
